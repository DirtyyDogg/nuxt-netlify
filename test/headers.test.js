const { appendHeaders, createHeadersContent } = require('../lib/utils')

describe('headers', () => {
  it('can append headers', () => {
    const headers1 = {
      '/*': [
        'X-Frame-Options: DENY',
        'X-XSS-Protection: 1; mode=block',
        'X-Content-Type-Options: nosniff'
      ]
    }

    const headers2 = {
      '/*': [
        'X-UA-Compatible: ie=edge',
        'Access-Control-Allow-Origin: *',
        'Referrer-Policy: origin'
      ]
    }

    const headers3 = {
      '/*': [
        'X-Frame-Options: SAMEORIGIN'
      ]
    }

    expect(appendHeaders(headers1, '/*', headers2['/*'])).toMatchSnapshot()
    expect(appendHeaders(headers1, '/*', headers3['/*'])).toMatchSnapshot()
  })

  it('can create header content', () => {
    const headers1 = [
      'Cache-Control: public, max-age=31536000, immutable'
    ]

    const headers2 = [
      'X-Frame-Options: DENY',
      'X-XSS-Protection: 1; mode=block'
    ]

    const headers3 = [
      'Basic-Auth: someuser:somepassword anotheruser:anotherpassword'
    ]

    const headers4 = [
      'Link: </style.css>; rel=preload; as=style',
      'Link: </main.js>; rel=preload; as=script',
      'Link: </image.jpg>; rel=preload; as=image'
    ]

    expect(appendHeaders(createHeadersContent(headers1, '/static/*'))).toMatchSnapshot()
    expect(appendHeaders(createHeadersContent(headers2, '/templates/index.html'))).toMatchSnapshot()
    expect(appendHeaders(createHeadersContent(headers3, '/something/*'))).toMatchSnapshot()
    expect(appendHeaders(createHeadersContent(headers4, '/*'))).toMatchSnapshot()
  })
})
