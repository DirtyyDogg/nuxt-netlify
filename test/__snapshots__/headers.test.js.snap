// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headers can append headers 1`] = `
Object {
  "/*": Array [
    "X-Frame-Options: DENY",
    "X-XSS-Protection: 1; mode=block",
    "X-Content-Type-Options: nosniff",
    "X-UA-Compatible: ie=edge",
    "Access-Control-Allow-Origin: *",
    "Referrer-Policy: origin",
  ],
}
`;

exports[`headers can append headers 2`] = `
Object {
  "/*": Array [
    "X-Frame-Options: DENY",
    "X-XSS-Protection: 1; mode=block",
    "X-Content-Type-Options: nosniff",
    "X-UA-Compatible: ie=edge",
    "Access-Control-Allow-Origin: *",
    "Referrer-Policy: origin",
    "X-Frame-Options: SAMEORIGIN",
  ],
}
`;

exports[`headers can create header content 1`] = `
"/static/*
  Cache-Control: public, max-age=31536000, immutable

"
`;

exports[`headers can create header content 2`] = `
"/templates/index.html
  X-Frame-Options: DENY
  X-XSS-Protection: 1; mode=block

"
`;

exports[`headers can create header content 3`] = `
"/something/*
  Basic-Auth: someuser:somepassword anotheruser:anotherpassword

"
`;

exports[`headers can create header content 4`] = `
"/*
  Link: </style.css>; rel=preload; as=style
  Link: </main.js>; rel=preload; as=script
  Link: </image.jpg>; rel=preload; as=image

"
`;
